src/ortho.rs

```diff
@@ -63,7 +117,7 @@ impl Ortho {
     
     fn get_forbidden(&self) -> Vec<u16> {
         let current_shell = self.get_current_shell();
-        let logical_coords = self.generate_logical_coordinates();
+        let logical_coords = get_logical_coordinates(&self.dimensions);
         
         self.storage.iter().enumerate()
```

Comment:
Cache the bits in here that deal with logical coordinate math. It should be possible to hit this cache and just get a list of coordinates in flat space for retrieval